qemu/tests/cpu_info_check.py

```diff
@@ -0,0 +1,134 @@
+                 "Haswell", "Haswell-noTSX", "IvyBridge",
+                 "SandyBridge", "Westmere", "Nehalem",
+                 "Penryn", "Conroe"]
+    cpu_type_re = {"EPYC-Rome": "rdpid,wbnoinvd,stibp,clwb,umip",
+                   "EPYC": "avx2,adx,bmi2,sha_ni",
+                   "Opteron_G5": "f16c,fma4,xop,tbm",
+                   "Opteron_G4": ("fma4,xop,avx,xsave,aes,sse4.2|sse4_2,"
+                                  "sse4.1|sse4_1,cx16,ssse3,sse4a"),
+                   "Opteron_G3": "cx16,sse4a",
+                   "Opteron_G2": "cx16",
+                   "Opteron_G1": "",
+                   "KnightsMill": "avx512_4vnniw,avx512pf,avx512er",
+                   "Icelake-Server": "avx512_vnni,la57,clflushopt",
+                   "Icelake-Client": ("avx512_vpopcntdq|avx512-vpopcntdq,"
+                                      "avx512vbmi,avx512_vbmi2|avx512vbmi2,"
+                                      "gfni,vaes,vpclmulqdq,avx512_vnni"),
+                   "Cascadelake-Server": ("avx512f,avx512dq,avx512bw,avx512cd,"
+                                          "avx512vl,clflushopt,avx512_vnni"),
+                   "Skylake-Server": "avx512f,clwb,xgetbv1,pcid",
+                   "Skylake-Client": "xgetbv1,pcid",
+                   "Broadwell": "adx,rdseed,3dnowprefetch,hle",
+                   "Broadwell-noTSX": "adx,rdseed,3dnowprefetch",
+                   "Haswell": "fma,avx2,movbe,hle",
+                   "Haswell-noTSX": "fma,avx2,movbe",
+                   "IvyBridge": "f16c,fsgsbase,erms",
+                   "SandyBridge": ("avx,xsave,aes,sse4_2|sse4.2,sse4.1|sse4_1,"
+                                   "cx16,ssse3"),
+                   "Westmere": "aes,sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                   "Nehalem": "sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                   "Penryn": "sse4.1|sse4_1,cx16,ssse3",
+                   "Conroe": "ssse3"}
+
+    qemu_path = utils_misc.get_qemu_binary(params)
+    qemu_version = env_process._get_qemu_version(qemu_path)
+    match = re.search(r'[0-9]+\.[0-9]+\.[0-9]+(\-[0-9]+)?',
+                      qemu_version)
+    host_qemu = match.group(0)
+    if host_qemu in VersionInterval('[3.1.0,)'):
+        cpu_types.extend(['Cascadelake-Server', 'Icelake-Client',
+                          'Icelake-Server', 'KnightsMill'])
+    if (host_qemu in VersionInterval('[4.2.0, )') and
+            params.get('os_variant') == 'rhel8'):
+        cpu_types.extend(
+            ['Cascadelake-Server-noTSX', 'Icelake-Client-noTSX',
+             'Icelake-Server-noTSX', 'Cascadelake-Server-v1',
+             'Cascadelake-Server-v2', 'Cascadelake-Server-v3',
+             'Icelake-Server-v1', 'Icelake-Server-v2', 'EPYC-v1', 'EPYC-v2'])
+        if params.get('os_app_variant') == '1':
+            cpu_types.extend(['Cooperlake', 'Cooperlake-v1', 'Snowridge',
+                              'Snowridge-v1', 'Snowridge-v2'])
+    if host_qemu in VersionInterval('[5.0.0, )'):
```

Comment:
since qemu will rebase to 5.1 we need to cover qemu 5.1.

Reply:
@nanliu-r This means >5.0, should I updated to >5.1, or 5.1 has new models introduced?

Reply:
> @nanliu-r This means >5.0, should I update to >5.1, or 5.1 has new models introduced?

since qemu has no final rebase version now qemu5.1 may add some new models, so we need to test both >5.0 and >5.1.
now please update to >5.0 and we can add the >5.1 later.

Reply:
@nanliu-r Current code is >5.0 already, thanks.